### PR TITLE
DOCSP-26526: v4.8.0 release updates

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-26526
Staging of Main Pages:
What's New - https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-26526-4.8-release/whats-new/#std-label-version-4.8
Upgrade Driver Versions - https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-26526-4.8-release/upgrade/#version-4.8-breaking-changes
Compat - https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-26526-4.8-release/compatibility/

Also check API links from TOC

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -35,6 +35,28 @@ The 4.8 driver ends support for MongoDB v3.4 and earlier. To learn
 more about breaking changes in this driver version and how to address 
 them, see :ref:`<java-server-release-change-v4.8>`.
 
+Behavioral changes with the 4.8 driver release include:
+
+- Specification of explicit dependency on the ``org.bson.codecs.record``
+  module if your application deploys the driver in an OSGi container and
+  relies on the driver for encoding and decoding Java records.
```
Comment (mongoKart) [inline source/whats-new.txt:42]: Issue: The way this is worded sounds like it's something the driver does for me, but my understanding is that it's now something I have to do 

Suggestion:

```suggestion
- Required explicit dependency on the ``org.bson.codecs.record``
  module if your application deploys the driver in an OSGi container and
  relies on the driver for encoding and decoding Java records.
```

File: source/whats-new.txt [hunk 2]
```diff
@@ -35,6 +35,28 @@ The 4.8 driver ends support for MongoDB v3.4 and earlier. To learn
...
+  optional dependency on both AWS SDK version 1 and version 2.
+- Support for obtaining credentials from the environment for both GCP
+  and Azure for in-use encryption.
+- Support for deserialization of records containing components with
+  generic parameters.
+- Support for deserialization of records containing components with more
+  specific container types, such as ``SortedSet``, ``NavigableMap``, ``LinkedList``,
+  or ``TreeSet``.
+- Support for recursive record types, meaning records whose type
+  definitions contain cycles.
```
Comment (mongoKart) [inline source/whats-new.txt:58]: Issue: "meaning" feels a little...casual? Clumsy?

Suggestion:

```suggestion
- Support for recursive record types, records whose type
  definitions contain cycles.
```

Or:

```suggestion
- Support for recursive record types (records whose type
  definitions contain cycles).
```

Or:

```suggestion
- Support for records whose type definitions contain cycles.
```

**Review by mongoKart — CHANGES_REQUESTED** [review]
A couple small changes. Overall looks good!

File: source/whats-new.txt (hunk 1 above)
Comment (rustagir) [reply source/whats-new.txt:42]: changed to reflect that user must do task

**Review by mongoKart — APPROVED** [review]
lgtm!

**Pull request merged**